test/e2e-go/cli/goal/expect/goalExpectCommon.exp

```diff
@@ -817,6 +817,20 @@ proc ::AlgorandGoal::CheckNetworkAddressForCors { NET_ADDRESS } {
     }
 }
 
+# Use curl to check if a network address supports private network access
+proc ::AlgorandGoal::CheckNetworkAddressForPNA { NET_ADDRESS } {
+    if { [ catch {
```

Comment:
why does it work? It should not with the `EnablePrivateNetworkAccessHeader` config set to `false` by default

Reply:
I was not able to make it to work using a private net either with or w/o `EnablePrivateNetworkAccessHeader` set in `config.json`. In both cases I have the same response
```
curl -X OPTIONS -H "Access-Control-Request-Private-Network: true" --head $(cat three-fut/Primary/algod-listen.net)
HTTP/1.1 404 Not Found
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Fri, 30 Aug 2024 19:05:02 GMT
Content-Length: 19
```

Reply:
> I was not able to make it to work using a private net either with or w/o `EnablePrivateNetworkAccessHeader` set in `config.json`. In both cases I have the same response
> 
> ```
> curl -X OPTIONS -H "Access-Control-Request-Private-Network: true" --head $(cat three-fut/Primary/algod-listen.net)
> HTTP/1.1 404 Not Found
> Content-Type: text/plain; charset=utf-8
> X-Content-Type-Options: nosniff
> Date: Fri, 30 Aug 2024 19:05:02 GMT
> Content-Length: 19
> ```

Hmm, I'm not able to replicate that response. I get a 204 even when the PNA config isn't enabled. But with it enabled get the header.
```sh
curl -X OPTIONS -H "Access-Control-Request-Private-Network: true" --head $(cat rootdir/node1/algod.net)
HTTP/1.1 204 No Content
Access-Control-Allow-Private-Network: true
Vary: Origin
Date: Mon, 02 Sep 2024 15:46:20 GMT
```

I've updated the test so it is correctly applying the config and have added a return error if it's not found. I'm new to using `expect` so please say if there's a better approach. Happy to put more time into this if needed.

Reply:
the check was not correct: there were no instructions in `expect{ }` block on what to do if not found. Usually it is done with eof check if there is no `exp_continue` and multi matching.